src/common/contracts/ao-process.ts

```diff
@@ -17,50 +17,32 @@
 import { connect } from '@permaweb/aoconnect';
 import { createData } from 'arbundles';
 
-import { AOContract, ContractSigner, Logger } from '../../types.js';
+import { AOContract, AoClient, ContractSigner, Logger } from '../../types.js';
 import { version } from '../../version.js';
+import { WriteInteractionError } from '../error.js';
 import { DefaultLogger } from '../logger.js';
 
 export class AOProcess implements AOContract {
   private logger: Logger;
   private processId: string;
-  private ao: {
-    result: any;
-    results: any;
-    message: any;
-    spawn: any;
-    monitor: any;
-    unmonitor: any;
-    dryrun: any;
-    assign: any;
-  };
+  private ao: AoClient;
 
   constructor({
     processId,
-    connectionConfig,
+    ao = connect(),
     logger = new DefaultLogger({ level: 'info' }),
   }: {
     processId: string;
-    connectionConfig?: {
-      CU_URL: string;
-      MU_URL: string;
-      GATEWAY_URL: string;
-      GRAPHQL_URL: string;
-    };
+    ao?: AoClient;
     logger?: DefaultLogger;
   }) {
     this.processId = processId;
     this.logger = logger;
-    this.ao = connect({
-      MU_URL: connectionConfig?.MU_URL,
-      CU_URL: connectionConfig?.CU_URL,
-      GATEWAY_URL: connectionConfig?.GATEWAY_URL,
-      GRAPHQL_URL: connectionConfig?.GRAPHQL_URL,
-    });
+    this.ao = ao;
   }
 
   // TODO: could abstract into our own interface that constructs different signers
-  async createAoSigner(
+  static async createAoSigner(
```

Comment:
why does this need to be static?

Reply:
When I was writing the spawnANT util, i needed to create a signer. I think we should make this static or a util - it doesnt actually use any of the class properties.

A WalletFactory class might be a good idea, so we can do like AOSignerFactory.arweave(ContractSigner) and AOSignerFactory.ethereum(ContractSigner)

Reply:
we can address in a separate PR